### PR TITLE
[RFC] be more tolerant with Modbus devices

### PR DESCRIPTION
**Pull request by jonasjelonek**
This PR originates from a problem I had and how I solved it. Some rough information is included in the commit message, but I will explain it here too:

In my home I have a Huawei SUN2000 inverter for my PV. It works fine so far, but in terms of accesses it outside of Huawei's own solution, it's a nightmare -.-. It provides a Modbus/TCP interface via its network connection, but it is quite awkward, e.g. it only allows one client at a time, but I have at least two appliances in my network that need to access it. Fortunately, I came across this modbus proxy app here :)

Unfortunately, this was running quite unstable, a lot of issue that the connection couldn't be established is gets lost after some time. I troubleshooted that and found out that the inverter and the modbus proxy don't work very well together, especially how they establish the connection.

TL;DR: The inverter needs some time between establishing the TCP connection and processing the first request. But the modbus proxy app does this rather lazily, it only connects right before the first request is sent. And on top, the inverter sometimes needs some retries for a request, but modbus-proxy-rs didn't do that really.

So I'm proposing some changes here.
[1] At first, I changed the lazy behaviour of modbus-proxy-rs to connecting to the Modbus device as soon as the first client devices connects to modbus-proxy-rs. After that, there is enough time for the client to wait a bit and then send its request.
[2] I introduced retries, since I experienced that quite often that a client connects to modbus-proxy-rs, sends a request but my inverter had problems with the connection or didn't process it properly. I chose an arbitrary try count of 5 here. After 5 attempts, it will fail completely but before it will just try again.
Both changes actually work very well, since I modified the code locally I had nearly no issues and my appliances can just connect to my inverter without any problems.

I would like to get some feedback on my changes, on the idea itself and on the code. I'm not expecting this to be merged as-is, but I would like to see that it somehow gets included.

### CODE REVIEW

**Comment by jonasjelonek** [comment]
@tiagocoutinho Any initial ideas on that? Currently this is kinda rough and bad style, I would definitely improve that in the next days.
